src/components/badge/index.tsx

```diff
@@ -27,14 +27,6 @@ const Badge = ({
 	const hasText = !!text
 	const hasLabel = !!ariaLabel
 	const isIconOnly = hasIcon && !hasText
-	const isStatusBadge =
-		color == 'success' || color == 'warning' || color == 'critical'
-
-	if (isStatusBadge && !hasIcon) {
-		throw new Error(
```

Comment:
Note: I've removed this conditional error, as I think it's quite a bit too prescriptive, and would otherwise prevent the changes in this PR. Rationale:

1. Status badges can use the text itself in addition to colour to communicate the necessary information. An icon should therefore not be _required_ to meet [use-of-color success criteria](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html).
2. The HDS badge documentation [explicitly mentions](https://helios.hashicorp.design/components/badge#icon) to "Use icons intentionally and only when they provide the user with extra value.". Having icons be _required_ for any badges that use status-oriented coloration seems to go against this recommendation.